webapp/src/app/shared/widgets/langlabel-widget/langlabel-widget.component.ts

```diff
@@ -0,0 +1,41 @@
+import { CdkTextareaAutosize } from '@angular/cdk/text-field';
+import {Component, ElementRef, QueryList, ViewChild, ViewChildren, ChangeDetectorRef} from '@angular/core';
+import {AbstractWidget} from '../abstract-widget';
+import {ResizedEvent} from 'angular-resize-event';
+
+@Component({
+	selector: 'app-langlabel-widget',
+	templateUrl: './langlabel-widget.component.html',
+	styleUrls: ['./langlabel-widget.component.scss', '../widget.scss']
+})
+export class LangLabelWidgetComponent extends AbstractWidget {
+	@ViewChildren(CdkTextareaAutosize) inputTextareas!: QueryList<CdkTextareaAutosize>;
+	languages: string[] = [
+		'en_US',
+		'de_DE',
+		'ja_JP',
+		'ko_KR',
+		'zh_CN',
+		'zh_TW'
+	];
+	
+	constructor() {
+		super();
+	}
+	
+	get hasValue(): boolean {
+		return this.settings[this.key] !== undefined;
+	}
+	
+	get defaultValue() {
+		const value = Object.create(null);
```

Comment:
why not `value = {}`? Is there any difference?

Reply:
The difference between `Object.create(null)` and `{}` is that if (for some reason) the object's prototype has been modified, the fields that have been added show up when iterating the object with a `for in` loop an object created with `{}`, while they don't if the object has been created with `Object.create(null)`.
It's probably better to iterate with `Object.keys` than with `for in` because of this, but I default to using `Object.create(null)` anyway when creating "data objects" just in case they are then iterated with a `for in` (and by planetary alignment the object prototype is also modified).

Reply:
I don't think this is relevant in the map editor since we neither modify the object prototype nor use `for in`. (As far as I remember - there might be some instances that snuck past me)

Reply:
So just use `{}` for simplicity?

Reply:
Yes, that would probably better. For consistency as well